src/installLogsPrinter.js

```diff
@@ -72,13 +75,20 @@ function installLogsPrinter(on, options = {}) {
         allMessages[data.spec][data.test] = messages;
```

Comment:
You will have to modify this object to be able to also contain the information whether a test has failed or not. You have this information on this level. And that's how you will know on print files which one you need to include and which not.